adapters/repos/db/index.go

```diff
@@ -753,12 +777,31 @@ func (i *Index) putObjectBatch(ctx context.Context, objects []*storobj.Object,
 	}
 
 	byShard := map[string]objsAndPos{}
+	// get all tenants shards
+	tenants := []string{}
+	tenantsStatus := map[string]string{}
+	var err error
+	for _, obj := range objects {
+		if obj.Object.Tenant == "" {
+			continue
+		}
+		tenants = append(tenants, obj.Object.Tenant)
+	}
+
+	if len(tenants) > 0 {
+		// TODO : remove any dupp
```

Comment:
question: do we want to tackle this in this PR ? Otherwise I feel like it should be `TODO-RAFT` to ensure we don't miss it later down the line

Reply:
sure, forgot about it , will take care of it